torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3593,6 +3593,27 @@ def _maybe_evaluate_static(
         symbols = list(expr.free_symbols)
 
         # Apply known runtime asserts
+        guards_exprs = []
+        for g in self.guards:
+            e = self.simplify(g.expr)
+            if compute_hint:
+                e = canonicalize_bool_expr(e.xreplace(self.var_to_val))
+            guards_exprs.append(e)
+
+        symbols_unbacked = symbols - self.var_to_val.keys()
```

Comment:
Isn't this set on symbols, aren't these slow?

Reply:
Yes, this is similar to that strange optimization, but profile did not show this line

Reply:
yeah, this is partially why I feel there was something very strange going on with that particular optimization.